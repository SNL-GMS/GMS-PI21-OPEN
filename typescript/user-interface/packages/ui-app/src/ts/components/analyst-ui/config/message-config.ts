import { CacheTypes } from '@gms/common-model';

export interface MessageConfig {
  historyFilters: Map<string, boolean>;
  invalidCellText: string;
  latestVersionCellText: string;
  tooltipMessages: {
    events: {
      associatedOpen: string;
      associatedComplete: string;
      associatedOther: string;
      unassociated: string;
    };
    history: {
      currentHistoryActionMessage: string;
      redoActionMessage: string;
      undoActionMessage: string;
      undoEventLevelActionMessage: string;
      redoEventLevelActionMessage: string;
      undoButtonAction: string;
      redoButtonAction: string;
    };
    location: {
      associatedOrCreatedMessage: string;
      locateCallInProgressMessage: string;
      rejectedOrUnassociatedMessage: string;
    };
    magnitude: {
      azimuthSourceToReceiverMessage: string;
      noStationsSetToDefiningMessage: string;
      setAllStationsNotDefiningMessage: string;
      setAllStationsDefiningMessage: string;
      sourceToReceiverAzimuthMessage: string;
      noAmplitudeMessage: string;
    };
    workflowConfirmation: {
      discardTooltip: string;
      cancelTooltip: string;
      header: string;
      text: string;
      title: string;
      cancelText: string;
      discardText: string;
    };
  };
  labels: {
    syncToWaveformDisplayVisibleTimeRange: string;
  };
}
export const messageConfig: MessageConfig = {
  // If true, filter out the keys from the history list
  historyFilters: new Map([
    [CacheTypes.UserActionDescription.UPDATE_EVENT_FROM_SIGNAL_DETECTION_CHANGE, true]
  ]),
  invalidCellText: 'Unknown',
  latestVersionCellText: 'Latest Version',
  tooltipMessages: {
    events: {
      associatedOpen: 'Associated to open event',
      associatedComplete: 'Associated to completed event',
      associatedOther: 'Associated to other event',
      unassociated: 'Unassociated'
    },
    history: {
      currentHistoryActionMessage: 'The current state displayed',
      redoActionMessage: 'Redo this action',
      undoActionMessage: 'Undo this action',
      undoEventLevelActionMessage: 'Revert event history',
      redoEventLevelActionMessage: 'Restore event history',
      undoButtonAction: 'Undo last action',
      redoButtonAction: 'Redo last undone action'
    },
    location: {
      associatedOrCreatedMessage: 'SD Associated or Created since last locate',
      locateCallInProgressMessage: 'Displays if a locate call is in progress',
      rejectedOrUnassociatedMessage: 'SD Rejected or Unassociated since last locate'
    },
    magnitude: {
      azimuthSourceToReceiverMessage: 'Source-to-Receiver Azimuth (\u00B0)',
      noStationsSetToDefiningMessage:
        'Select at least one defining station to calculate network magnitude',
      setAllStationsNotDefiningMessage: 'Set all stations as not defining',
      setAllStationsDefiningMessage: 'Set all stations as defining',
      sourceToReceiverAzimuthMessage: 'Source to Receiver Azimuth (\u00B0)',
      noAmplitudeMessage: 'Make amplitude measurement to set as defining'
    },
    workflowConfirmation: {
      discardTooltip: 'Discard your changes and open a new interval',
      cancelTooltip: 'Cancel and do not open a new interval',
      title: 'Warning',
      header: 'You have unsaved changes in your workspace.',
      text: ' Do you want to discard these changes and load data for a different time range?',
      cancelText: 'Cancel',
      discardText: 'Discard my changes'
    }
  },
  labels: {
    syncToWaveformDisplayVisibleTimeRange: 'Sync to Waveform Display visible time range'
  }
};
