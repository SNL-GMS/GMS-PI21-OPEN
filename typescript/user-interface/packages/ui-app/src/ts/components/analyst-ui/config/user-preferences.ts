import { CommonTypes, LegacyEventTypes } from '@gms/common-model';
import type { AnalystWorkspaceTypes } from '@gms/ui-state';

import { semanticColors } from '~scss-config/color-preferences';

import { QcMaskCategory } from './system-config';

export interface MaskDisplayFilter {
  color: string;
  visible: boolean;
  name: string;
}

export interface QcMaskDisplayFilters {
  ANALYST_DEFINED: MaskDisplayFilter;
  CHANNEL_PROCESSING: MaskDisplayFilter;
  DATA_AUTHENTICATION: MaskDisplayFilter;
  REJECTED: MaskDisplayFilter;
  STATION_SOH: MaskDisplayFilter;
  WAVEFORM_QUALITY: MaskDisplayFilter;
}
export interface AnalystWorkSpaceColorPreferences {
  maskDisplayFilters: QcMaskDisplayFilters;
}

export const analystWorkspaceColors: AnalystWorkSpaceColorPreferences = {
  maskDisplayFilters: {
    ANALYST_DEFINED: {
      color: semanticColors.qcAnalystDefined,
      visible: true,
      name: QcMaskCategory.ANALYST_DEFINED
    },
    CHANNEL_PROCESSING: {
      color: semanticColors.qcChannelProcessing,
      visible: true,
      name: QcMaskCategory.CHANNEL_PROCESSING
    },
    DATA_AUTHENTICATION: {
      color: semanticColors.qcDataAuthentication,
      visible: true,
      name: QcMaskCategory.DATA_AUTHENTICATION
    },
    REJECTED: {
      color: semanticColors.qcRejected,
      visible: false,
      name: QcMaskCategory.REJECTED
    },
    STATION_SOH: {
      color: semanticColors.qcStationSOH,
      visible: true,
      name: QcMaskCategory.STATION_SOH
    },
    WAVEFORM_QUALITY: {
      color: semanticColors.qcWaveformQuality,
      visible: true,
      name: QcMaskCategory.WAVEFORM_QUALITY
    }
  }
};

/** Initial Mag type */
const initialMagType: AnalystWorkspaceTypes.DisplayedMagnitudeTypes = {};
initialMagType[LegacyEventTypes.MagnitudeType.MB] = true;
initialMagType[LegacyEventTypes.MagnitudeType.MBMLE] = true;
initialMagType[LegacyEventTypes.MagnitudeType.MS] = true;
initialMagType[LegacyEventTypes.MagnitudeType.MSMLE] = true;
Object.freeze(initialMagType);
export interface UserPreferences {
  azimuthSlowness: {
    defaultLead: number;
    defaultLength: number;
    defaultStepSize: number;
    // Minimum height and width of fk rendering
    minFkLengthPx: number;
    // Maximum height and width of fk rendering
    maxFkLengthPx: number;
  };
  initialMagType: AnalystWorkspaceTypes.DisplayedMagnitudeTypes;
  colors: AnalystWorkSpaceColorPreferences;
  map: {
    icons: {
      event: string;
      eventScale: number;
      station: string;
      stationScale: number;
      scaleFactor: number;
      displayDistance: number;
      pixelOffset: number;
    };
    widths: {
      unselectedSignalDetection: number;
      selectedSignalDetection: number;
    };
    defaultTo3D: boolean;
  };
  defaultSignalDetectionPhase: CommonTypes.PhaseType;
  signalDetectionList: {
    autoFilter: boolean;
    showIds: boolean;
  };
  eventList: {
    showIds: boolean;
    showAssocIds: boolean;
  };
  location: {
    preferredLocationSolutionRestraintOrder: LegacyEventTypes.DepthRestraintType[];
  };
  list: {
    minWidthPx: number;
    widthOfTableMarginsPx: number;
  };
  distanceUnits: CommonTypes.DistanceUnits;
}
export const userPreferences: UserPreferences = {
  azimuthSlowness: {
    defaultLead: 1,
    defaultLength: 4,
    defaultStepSize: 5,
    // Minimum height and width of fk rendering
    minFkLengthPx: 265,
    // Maximum height and width of fk rendering
    maxFkLengthPx: 500
  },
  colors: analystWorkspaceColors,
  initialMagType,
  map: {
    icons: {
      event: 'circle-transition.png',
      eventScale: 0.07,
      station: 'outlined-triangle.png',
      stationScale: 0.12,
      scaleFactor: 1.5,
      displayDistance: 1e6,
      pixelOffset: 15
    },
    widths: {
      unselectedSignalDetection: 1,
      selectedSignalDetection: 3
    },
    defaultTo3D: false
  },
  defaultSignalDetectionPhase: CommonTypes.PhaseType.P,
  signalDetectionList: {
    autoFilter: true,
    showIds: true
  },
  eventList: {
    showIds: true,
    showAssocIds: false
  },
  location: {
    preferredLocationSolutionRestraintOrder: [
      LegacyEventTypes.DepthRestraintType.UNRESTRAINED,
      LegacyEventTypes.DepthRestraintType.FIXED_AT_SURFACE,
      LegacyEventTypes.DepthRestraintType.FIXED_AT_DEPTH
    ]
  },
  list: {
    minWidthPx: 60,
    widthOfTableMarginsPx: 16
  },
  distanceUnits: CommonTypes.DistanceUnits.degrees
};
