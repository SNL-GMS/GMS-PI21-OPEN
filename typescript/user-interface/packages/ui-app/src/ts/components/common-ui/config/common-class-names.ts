export interface CommonClassNames {
  sharedTableClasses: string;
}

export const commonClassNames: CommonClassNames = {
  sharedTableClasses: `
    with-separated-rows
    ag-theme-dark
    ag-theme-dark--soh`
};
