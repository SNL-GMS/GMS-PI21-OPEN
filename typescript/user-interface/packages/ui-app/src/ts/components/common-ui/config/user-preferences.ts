export interface UserPreferences {
  baseSoundsPath: string;
  tableRowHeightPx: number;
  tableHeaderHeightPx: number;
  soundFileNotFound(filename: string): string;
  configuredAudibleNotificationFileNotFound(filename: string): string;
}

export const userPreferences: UserPreferences = {
  baseSoundsPath: `./resources/sounds/`,
  tableRowHeightPx: 36,
  tableHeaderHeightPx: 18,
  soundFileNotFound: (filename: string) => `Failed to load sound file "${filename}".`,
  configuredAudibleNotificationFileNotFound: (filename: string) =>
    `Failed to load sound file "${filename}" for configured audible notification. Sound will not play.`
};
