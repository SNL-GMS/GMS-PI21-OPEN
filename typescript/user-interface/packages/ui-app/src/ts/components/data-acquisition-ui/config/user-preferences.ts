import type { HorizontalDividerTypes } from '@gms/ui-core-components';

import { semanticColors } from '~scss-config/color-preferences';
import { gmsLayout } from '~scss-config/layout-preferences';

export interface DataAcquisitionUserPreferences {
  readonly colors: {
    readonly ok: string;
    readonly warning: string;
    readonly strongWarning: string;
    readonly none: string;
  };
  readonly overviewMinContainerRange: HorizontalDividerTypes.HorizontalDividerSizeRange;
  readonly stationStatisticsMinContainerRange: HorizontalDividerTypes.HorizontalDividerSizeRange;
  readonly sohStatusUpdateTime: number;
  readonly transferredFilesGapsUpdateTime: number;
  readonly tableRowHeightPx: number;
  readonly defaultOverviewGroupHeight: number;
  readonly minChartHeightPx: number;
  readonly minChartWidthPx: number;
}

export const dataAcquisitionUserPreferences: DataAcquisitionUserPreferences = {
  colors: {
    ok: semanticColors.dataAcqOk,
    warning: semanticColors.dataAcqWarning,
    strongWarning: semanticColors.dataAcqStrongWarning,
    none: semanticColors.dataAcqNone
  },
  overviewMinContainerRange: {
    minimumTopHeightPx: 145,
    minimumBottomHeightPx: 110
  },
  stationStatisticsMinContainerRange: {
    minimumTopHeightPx: 145,
    minimumBottomHeightPx: 110
  },
  sohStatusUpdateTime: 60000,
  transferredFilesGapsUpdateTime: 600000,
  tableRowHeightPx: 36,
  defaultOverviewGroupHeight: 360,
  minChartWidthPx: gmsLayout.minChartWidthPx,
  minChartHeightPx: gmsLayout.minChartHeightPx
};
