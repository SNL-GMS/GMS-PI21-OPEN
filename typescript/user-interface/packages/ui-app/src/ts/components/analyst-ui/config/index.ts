import { ConfigurationTypes } from '@gms/common-model';

import type { EnvironmentConfig } from './environment-config';
import { environmentConfig } from './environment-config';
import type { SystemConfig } from './system-config';
import { systemConfig } from './system-config';
import type { UserPreferences } from './user-preferences';
import { userPreferences } from './user-preferences';

export interface AnalystUiConfig {
  userPreferences: UserPreferences;
  environment: EnvironmentConfig;
  systemConfig: SystemConfig;
  unassociatedSDColor: string;
}

export const analystUiConfig: AnalystUiConfig = {
  userPreferences,
  environment: environmentConfig,
  systemConfig,
  unassociatedSDColor: ConfigurationTypes.defaultColorTheme.unassociatedSDColor
};
export { EnvironmentConfig, environmentConfig } from './environment-config';
export { SystemConfig, systemConfig } from './system-config';
export { QcMaskDisplayFilters, UserPreferences, userPreferences } from './user-preferences';
