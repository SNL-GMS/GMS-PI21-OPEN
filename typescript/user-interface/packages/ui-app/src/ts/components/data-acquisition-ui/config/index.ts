import type { DataAcquisitionUserPreferences } from './user-preferences';
import { dataAcquisitionUserPreferences } from './user-preferences';

export interface DataAcquisitionUIConfig {
  dataAcquisitionUserPreferences: DataAcquisitionUserPreferences;
}

export const dataAcquisitionUIConfig: DataAcquisitionUIConfig = {
  dataAcquisitionUserPreferences
};

export { dataAcquisitionUserPreferences } from './user-preferences';
