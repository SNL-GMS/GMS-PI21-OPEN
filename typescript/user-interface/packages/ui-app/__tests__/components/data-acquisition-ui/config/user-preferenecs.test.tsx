import { dataAcquisitionUserPreferences } from '../../../../src/ts/components/data-acquisition-ui/config/user-preferences';

// set up window alert and open so we don't see errors
(window as any).alert = jest.fn();
(window as any).open = jest.fn();

describe('User Preferences Config', () => {
  it('is exported', () => {
    expect(dataAcquisitionUserPreferences).toBeDefined();
  });

  it('is config defined', () => {
    expect(JSON.stringify(dataAcquisitionUserPreferences)).toMatchSnapshot();
  });
});
