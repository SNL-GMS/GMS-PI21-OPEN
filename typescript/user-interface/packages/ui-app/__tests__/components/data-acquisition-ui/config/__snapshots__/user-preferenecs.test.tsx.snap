// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Preferences Config is config defined 1`] = `"{\\"colors\\":{\\"ok\\":\\"#fff\\",\\"warning\\":\\"#fff\\",\\"strongWarning\\":\\"#fff\\"},\\"overviewMinContainerRange\\":{\\"minimumTopHeightPx\\":145,\\"minimumBottomHeightPx\\":110},\\"stationStatisticsMinContainerRange\\":{\\"minimumTopHeightPx\\":145,\\"minimumBottomHeightPx\\":110},\\"sohStatusUpdateTime\\":60000,\\"transferredFilesGapsUpdateTime\\":600000,\\"tableRowHeightPx\\":36,\\"defaultOverviewGroupHeight\\":360}"`;
